src/types/response.rs

```diff
@@ -217,14 +224,64 @@ impl QueryResponse {
                 .ok_or(CruxError::ParseEdnError(format!(
                     "The following Edn cannot be parsed to BTreeSet: {:?}",
                     edn
-                )))
-                .unwrap()
+                )))?
                 .map(|e| e.to_vec().unwrap())
                 .collect::<BTreeSet<Vec<String>>>())
         }
     }
 }
 
+#[cfg(feature = "async")]
+#[derive(Clone, Debug, PartialEq)]
+/// When feature `async` is enabled this is the response type for endpoint `/query`.
+pub struct QueryAsyncResponse(BTreeSet<Vec<String>>);
+
+#[cfg(feature = "async")]
+impl QueryAsyncResponse {
+    pub(crate) fn deserialize(resp: String) -> QueryAsyncResponse {
+        let edn = from_str(&resp.clone()).unwrap();
+        if edn.set_iter().is_some() {
+            QueryAsyncResponse {
+                0: edn
+                    .set_iter()
+                    .ok_or(CruxError::ParseEdnError(format!(
+                        "The following Edn cannot be parsed to BTreeSet: {:?}",
+                        edn
+                    )))
+                    .unwrap()
```

Comment:
Shouldn't we return `Result<QueryAsyncResponse, CruxError>` just like the sync version?

Reply:
can't, problem I said before

Reply:
Okay!